dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/datafile/DataFile.java

```diff
@@ -339,11 +340,24 @@ public void setIngestRequest(IngestRequest ingestRequest) {
 
     public String getIngestReportMessage() {
         if (ingestReports != null && ingestReports.size() > 0) {
-            if (ingestReports.get(0).getReport() != null && !"".equals(ingestReports.get(0).getReport())) {
-                return ingestReports.get(0).getReport();
+            IngestReport ingestReport = ingestReports.get(0);
+            String report = ingestReport.getReport();
+            if (!StringUtils.isEmpty(report)) {
+
+                if (ingestReport.getArgumentsBundleKey().isDefined() && ingestReport.getReportArguments().isEmpty()){
+                    String bundleValue = BundleUtil.getStringFromBundle(ingestReport.getArgumentsBundleKey().get());
+
+                    return bundleValue.isEmpty() ? report : bundleValue;
+                } else if (ingestReport.getArgumentsBundleKey().isDefined() && !ingestReport.getReportArguments().isEmpty()) {
+                    String bundleValue = BundleUtil.getStringFromBundle(ingestReport.getArgumentsBundleKey().get(), ingestReport.getReportArguments());
+
+                    return bundleValue.isEmpty() ? report : bundleValue;
+                }
+
+                return report;
```

Comment:
Feature envy - this whole logic seems better placed in `IngestReport` class instead.

Reply:
I agree, I kinda wanted to stop myself from refactoring overall since it is very easy to sink into "Shotgun Surgery".

Reply:
All that code was actually added here now and "extract method" refactoring is far from "shotgun surgery". Let's move this logic where it belongs.